arcade/sound.py

```diff
@@ -108,13 +108,14 @@ def stop(self, player: media.Player) -> None:
         if player in media.Source._players:
             media.Source._players.remove(player)
 
-    def get_length(self) -> float:
+    def get_length(self) -> Optional[float]:
```

Comment:
If this is optional, we should probably update the docs to note when we'd get None back.

Reply:
Should we also update the docs for `is_complete` to say that it will throw an exception if you call it on a streaming source?

Reply:
I based this on a comment in pyglet's code:

https://github.com/pyglet/pyglet/blob/515d7a52a63fc20d76109f06a02f059459ec03b8/pyglet/media/codecs/base.py#L207-L216

> Not all source durations can be determined; in this case the value is ``None``.

But I'm not sure when that happens, and I'm not sure how to test it.  If I load a sound and get a duration, it does not mean that *other* OSes or codecs won't return `None`.

Reply:
After some Discord discussion, it sounds easier to assert that duration is always available, and throw an exception if not.